src/net_processing.cpp

```diff
@@ -186,6 +186,53 @@ struct QueuedBlock {
     std::unique_ptr<PartiallyDownloadedBlock> partialBlock;
 };
 
+class TimeOffsets
+{
+private:
+    // Historic maximum number of time offsets we include in our median.
+    static constexpr size_t N{199};
+
+    mutable Mutex m_mutex;
+    std::array<int64_t, N> m_offsets GUARDED_BY(m_mutex){};
```

Comment:
Wdyt about making the offsets typesafe with `std::chrono::seconds`?

<details>
<summary>git diff on 8b1c133921</summary>

```diff
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 664992dcab..4a7243b82f 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -193,12 +193,12 @@ private:
     static constexpr size_t N{199};
 
     mutable Mutex m_mutex;
-    std::array<int64_t, N> m_offsets GUARDED_BY(m_mutex){};
+    std::array<std::chrono::seconds, N> m_offsets GUARDED_BY(m_mutex){};
     size_t m_index GUARDED_BY(m_mutex){0};
 
 public:
     /** Add a new time offset sample. */
-    void Add(int64_t offset) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    void Add(std::chrono::seconds offset) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
     {
         AssertLockNotHeld(m_mutex);
         LOCK(m_mutex);
@@ -213,15 +213,15 @@ public:
     }
 
     /** Compute and return the median of the collected time offset samples. */
-    int64_t Median() const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    std::chrono::seconds Median() const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
     {
         AssertLockNotHeld(m_mutex);
         LOCK(m_mutex);
 
         // Only compute the median from 5 or more samples.
-        if (m_index < 4) return 0;
+        if (m_index < 4) return std::chrono::seconds{0};
 
-        std::array<int64_t, N> sorted_copy = m_offsets;
+        std::array<std::chrono::seconds, N> sorted_copy = m_offsets;
         std::sort(sorted_copy.begin(), sorted_copy.end());
 
         if (m_index % 2 == 0) {
@@ -3621,11 +3621,11 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
 
         peer->m_time_offset = nTime - GetTime();
         if (!pfrom.IsInboundConn()) {
-            m_outbound_time_offsets.Add(peer->m_time_offset);
+            m_outbound_time_offsets.Add(std::chrono::seconds{peer->m_time_offset});
 
             auto median = m_outbound_time_offsets.Median();
-            if (median > m_opts.time_offset_warn_threshold.count() ||
-                median < -m_opts.time_offset_warn_threshold.count()) {
+            if (median > m_opts.time_offset_warn_threshold ||
+                median < -m_opts.time_offset_warn_threshold) {
                 SetMedianTimeOffsetWarning();
             }
         }
diff --git a/src/net_processing.h b/src/net_processing.h
index 0fec728ebb..4fb280ed16 100644
--- a/src/net_processing.h
+++ b/src/net_processing.h
@@ -46,7 +46,7 @@ struct CNodeStateStats {
 };
 
 struct PeerManagerInfo {
-    int64_t median_outbound_time_offset{0};
+    std::chrono::seconds median_outbound_time_offset{0};
     bool ignores_incoming_txs{false};
 };
 
diff --git a/src/rpc/net.cpp b/src/rpc/net.cpp
index ea53a974f1..780f1383ed 100644
--- a/src/rpc/net.cpp
+++ b/src/rpc/net.cpp
@@ -673,7 +673,7 @@ static RPCHelpMan getnetworkinfo()
     if (node.peerman) {
         auto peerman_info{node.peerman->GetInfo()};
         obj.pushKV("localrelay", !peerman_info.ignores_incoming_txs);
-        obj.pushKV("timeoffset", peerman_info.median_outbound_time_offset);
+        obj.pushKV("timeoffset", peerman_info.median_outbound_time_offset.count());
     }
     if (node.connman) {
         obj.pushKV("networkactive", node.connman->GetNetworkActive());

```
</details>

Reply:
Good idea, but I'd prefer to leave that for a follow up

Reply:
Care to elaborate why? Since this is newly introduced code, seems sensible to do it right away?